api-report/telemetry-utils.api.md

```diff
@@ -101,6 +101,8 @@ export interface IConfigProvider extends IConfigProviderBase {
 export interface IConfigProviderBase {
     // (undocumented)
     getRawConfig(name: string): ConfigTypes;
+    // (undocumented)
+    getRawConfigEntries?(): [string, string][];
```

Comment:
this worries me a bit. config providers may not know all config, or they may require network round trips. I wonder if we could do something like log on read, where whenever a config is read, we log the value the first time it is read.

Reply:
Ok, I can try something like that.

Reply:
Is there potentially a better solution? We do a lot of config option reads especially when we run the startup.

Reply:
Also, if that's the case, we would already be making a lot of network calls to get all these various config options, this seems like a much bigger issue if we make a network call each time we try to get a single config option.

Reply:
see my comment below. the current solution won't work for our trivial configProvider.

Reply:
In the container runtime alone, we would have potentially made 5 network calls just on load alone.

Reply:
i would only log defined values, i wouldn't log undefined. its not quite as easy to work with, but i would expect any particulary session only has a few configs in use at any one time.

Reply:
i think you are right that network call is unlikley. the other problems remain.

Reply:
If i think about some of the implementations, i think they sometimes kick off a lazy network call, and refresh the local cache. if a config is read before the call, they get the cache value for the session. if they read the config after the call, they could get the new value. this change would pin all the version upfront.